serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/ObjectInspectorUtils.java

```diff
@@ -1652,4 +1654,20 @@ public static boolean hasAllFieldsSettable(ObjectInspector oi,
   private ObjectInspectorUtils() {
     // prevent instantiation
   }
+
+  /**
+   * Returns slot value used for ordering the fields to make it deterministic
+   * @param field : field of a given class
+   * @return
+   */
+  private static int getSlotValue(Field field) {
+    try {
+      Field slotField = Field.class.getDeclaredField("slot");
+      slotField.setAccessible(true);
+      return slotField.getInt(field);
+    } catch (NoSuchFieldException | IllegalAccessException | IllegalArgumentException e) {
+      LOG.info("Field not found: {}", e);
```

Comment:
I think this will not log properly.
These braces '{}' are used when you want to format the log message substituting them with provided values, but the second element here is not a string, but an Exception, so it will log an exception with the message "Field not found: {}" as is.

Reply:
Hi, 
Thanks for the review. Understood. I have made the changes to log the error. Please let me know if this looks good.

Reply:
Yes, just I think it would be better to make the exception message something like "Error getting a slot number
If an exception happens here, why do we log it on info level and return 0?
In Comparator zero means equality. 
Maybe it's better to log it on error level and throw a Hive-specific runtime exception?

Reply:
Agreed, basically the change is to make the function deterministic so that the mentioned tests are not flaky anymore. As you said, returning 0 can still return random order as it means equality. 
I'll have changed this to throw a new RuntimeException(similar to the exceptions used in other parts of the class) with the same error message.